### ISSUE TITLE
Warning image [...] is not used.

### ISSUE THREAD

**Issue by Thom-x** [GoogleContainerTools/skaffold#9160]
### Expected behavior

No warnings about image not deployed and debug/log tail working.

### Actual behavior

When I'm running a deployment on EKS:

```
skaffold debug --filename skaffold-stackator.yaml --label ide=cloudcodeintellij --cleanup=false --rpc-port 50051 --port-forward=true --wait-for-deletions-max=2m0s --status-check=true --module myModule --verbosity info
```

I have the following warning:

```
image [myregistry/myimage:cc2b538ed799f4a265da9ff3ffd4ad94945caf0cca05f8136d3f5fb253acf1ee@sha256:dbf2dc81ceee4e4a7809c52cc8593bd98f4ad0f399e33219e8e1786721c79b90] is not used.
See helm documentation on how to replace image names with their actual tags: https://skaffold.dev/docs/pipeline-stages/deployers/helm/#image-configuration
```

```
Validating Kubernetes dependencies...
Validating image repository settings...
Validating Skaffold module settings...
Caution: deploying a subset of modules may result in errors if there are 
	 implicit dependencies not expressed in the skaffold.yaml configuration files.

/home/xxx/.cache/cloud-code/bin/versions/16a62023a2e948d1eec96707bf21410817d08431ca8488a962a5cce9e71819df/skaffold debug --filename skaffold-stackator.yaml --label ide=cloudcodeintellij --cleanup=false --rpc-port 50051 --port-forward=true --wait-for-deletions-max=2m0s --status-check=true --module myModule --verbosity info
starting gRPC server on port 50051
Skaffold &{Version:v2.7.1 ConfigVersion:skaffold/v4beta6 GitVersion: GitCommit:4557ab1d4c8361977bbade432169f0aa048f4310 BuildDate:2023-09-13T14:43:49Z GoVersion:go1.21.0 Compiler:gc Platform:linux/amd64 User:}
Loaded Skaffold defaults from "/home/xxx/.skaffold/config"
map entry found when executing locate for &{myregistry/myimage /home/xxx/xxx <nil> {<nil> <nil> <nil> 0xc00079df40 <nil> <nil> <nil>} [] {[] []} [] } of type *latest.Artifact and pointer: 824640310528
Using kubectl context: aws_dev
no kpt renderer or deployer found, skipping hydrated-dir creation
build concurrency first set to 1 parsed from *local.Builder[0]
final build concurrency value is 1
Generating tags...
 - myregistry/myimage -> myregistry/myimage:cc2b538ed799f4a265da9ff3ffd4ad94945caf0cca05f8136d3f5fb253acf1ee
Tags generated in 4.319 seconds
Checking cache...
 - myregistry/myimage: Found Remotely
Cache check completed in 1.295 second
Starting render...
Render completed in 86.215µs
Tags used in deployment:
 - myregistry/myimage -> myregistry/myimage:cc2b538ed799f4a265da9ff3ffd4ad94945caf0cca05f8136d3f5fb253acf1ee@sha256:dbf2dc81ceee4e4a7809c52cc8593bd98f4ad0f399e33219e8e1786721c79b90
Starting deploy...
Deploying with helm v3.13.1 ...
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /xxx.config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /xxx.config
Release "myrelease" has been upgraded. Happy Helming!
NAME: myrelease
LAST DEPLOYED: Thu Nov  2 16:36:57 2023
NAMESPACE: stack-tmaugin
STATUS: deployed
REVISION: 2
TEST SUITE: None
error parsing object 0 from string: error decoding parsed yaml: yaml: mapping values are not allowed in this context
error parsing object 3 from string: error decoding parsed yaml: no kind "PrometheusRule" is registered for version "monitoring.coreos.com/v1" in scheme "k8s.io/client-go/kubernetes/scheme/register.go:79"
error parsing object 4 from string: error decoding parsed yaml: no kind "SealedSecret" is registered for version "bitnami.com/v1alpha1" in scheme "k8s.io/client-go/kubernetes/scheme/register.go:79"

image [myregistry/myimage:cc2b538ed799f4a265da9ff3ffd4ad94945caf0cca05f8136d3f5fb253acf1ee@sha256:dbf2dc81ceee4e4a7809c52cc8593bd98f4ad0f399e33219e8e1786721c79b90] is not used.
See helm documentation on how to replace image names with their actual tags: https://skaffold.dev/docs/pipeline-stages/deployers/helm/#image-configuration

Waiting for deployments to stabilize...
 - xxx: waiting for rollout to finish: 1 old replicas are pending termination...
 - xxx: running.
 - xxx is ready.
Deployments stabilized in 2 minutes 13.507 seconds
Deploy completed in 2 minutes 29.201 seconds
Port forwarding service/xxx in namespace xxx, remote port 80 -> http://127.0.0.1:4504
Port forwarding service/xxx in namespace xxx, remote port 81 -> http://127.0.0.1:4503
Listing files to watch...
 - xxx
List generated in 1.516952ms
Press Ctrl+C to exit
Not watching for changes...
```

I checked the deployment:

```
image: myregistry/myimage:cc2b538ed799f4a265da9ff3ffd4ad94945caf0cca05f8136d3f5fb253acf1ee@sha256:dbf2dc81ceee4e4a7809c52cc8593bd98f4ad0f399e33219e8e1786721c79b90
```

The pod itself:

```
Container ID: containerd://950b2cfb5b55ec83b783b6a8425d3c00b4cd43467361f95d2cabd16704ed8821
Image: myregistry/myimage:cc2b538ed799f4a265da9ff3ffd4ad94945caf0cca05f8136d3f5fb253acf1ee@sha256:dbf2dc81ceee4e4a7809c52cc8593bd98f4ad0f399e33219e8e1786721c79b90
Image ID: myregistry/myimage@sha256:dbf2dc81ceee4e4a7809c52cc8593bd98f4ad0f399e33219e8e1786721c79b90
```

### Information

- Skaffold version: 2.7.1
- Operating system: Ubuntu 23.10 
- Installed via: Google Cloud SDK
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v4beta6
kind: Config
metadata:
    name: myModule
build:
    tagPolicy:
        inputDigest: {}
    artifacts:
        -   image: myregistry/myimage
            context: /mycontext
            jib:
                type: maven
                project: project:module
                fromImage: mybaseimage
                args:
                    - "-DskipTests=true"
                    - "-Dpmd.skip=true"
                    - "-Dcpd.skip=true"
                    - "-Dcheckstyle.skip=true"
                    - "-DbuildTimestamp=0"
                    - "-Djib.containerizingMode=exploded" # allow to watch file change
deploy:
    helm:
        releases:
            -   name: myrelease
                remoteChart: repoChart
                upgradeOnChange: true
                version: 0.10.1
                namespace: namespace
                createNamespace: false
                valuesFiles:
                    - values.yaml
                setValueTemplates:
                    image.tag: "{{.IMAGE_TAG_myregistry_myimage}}@{{.IMAGE_DIGEST_myregistry_myimage}}"
                setValues:
                    image.pullPolicy: "IfNotPresent"
```

After the deployment is done, debug port is not forwarded, and log tailing is also not working (nothing show up after).
I'm also doing a similar config on a minikube cluster, and it works fine (probably because of local build vs pushing the image).

I tried all possibles tag policies, and used `setValueTemplates` with all possibles combinations of tag+digest being present or not.

**Comment by Thom-x** [comment]
**TL;DR**
`chmod 600 /path/to/my/kubeconfig`

I finally found the solution !
No idea why, but I had the warning:

```
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /xxx.config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /xxx.config
```

Fixed it with proper permissions and now it runs fine. Any idea why ?
It's absolutely not clear to me why such permission on kubeconfig is responsible for the image not being found.

**Comment by mecampbellsoup** [comment]
@Thom-x that was really all it took? I see this warning/error constantly...

**Comment by mecampbellsoup** [comment]
NGL I can't believe this is a persistent issue in Skaffold. Someone on our team gets totally lost by this seemingly every month or so (really it's an "every new hire" thing) and comes to us asking, "How come the skaffold process isn't... _doing_ anything?"
